components/micro-gateway-core/src/main/ballerina/src/gateway/utils/throttle_util.bal

```diff
@@ -57,54 +57,52 @@ public function isRequestThrottled(string key) returns [boolean, boolean] {
     printDebug(KEY_THROTTLE_UTIL, "throttle data key : " + key);
     boolean isThrottled = throttleDataMap.hasKey(key);
     if (isThrottled) {
-        int currentTime = time:currentTime().time;
-        GlobalThrottleStreamDTO dto = <GlobalThrottleStreamDTO>throttleDataMap[key];
-        int timeStamp = dto.expiryTimeStamp;
+        ThrottledRequest dto = <ThrottledRequest>throttleDataMap[key];
         boolean stopOnQuota = dto.stopOnQuota;
         if (enabledGlobalTMEventPublishing == true) {
             stopOnQuota = true;
-        }
-        if (timeStamp >= currentTime) {
             return [isThrottled, stopOnQuota];
-        } else {
-            var value = throttleDataMap.remove(key);
-            return [false, stopOnQuota];
         }
+        return [isThrottled, stopOnQuota];
     }
     return [isThrottled, false];
 }
 
-public function publishNonThrottleEvent(RequestStreamDTO throttleEvent) {
+public function publishNonThrottleEvent(InputRequest throttleEvent) {
     //Publish throttle event to traffic manager
     if (enabledGlobalTMEventPublishing == true) {
         publishThrottleEventToTrafficManager(throttleEvent);
         printDebug(KEY_THROTTLE_UTIL, "Throttle out event is sent to the traffic manager.");
     }
     //Publish throttle event to internal policies
     else {
-        requestStream.publish(throttleEvent);
-        printDebug(KEY_THROTTLE_UTIL, "Request stream : " + requestStream.toString());
+        inputStream.publish(throttleEvent);
+        printDebug(KEY_THROTTLE_UTIL, "Request stream : " + inputStream.toString());
         printDebug(KEY_THROTTLE_UTIL, "Throttle out event is sent to the queue.");
     }
 }
 
 public function initializeThrottleSubscription() {
-    globalThrottleStream.subscribe(onReceiveThrottleEvent);
+    outputStream.subscribe(onReceiveThrottleEvent);
     isStreamsInitialized = true;
     printDebug(KEY_THROTTLE_UTIL, "Successfully subscribed global throttle stream.");
 }
 
-//public function getInitThrottleSubscriptionFuture() returns future<()>{
-//return ftr;
-//}
-public function onReceiveThrottleEvent(GlobalThrottleStreamDTO throttleEvent) {
-    printDebug(KEY_THROTTLE_UTIL, "Event GlobalThrottleStream: throttleKey: " + throttleEvent.throttleKey +
-    " ,isThrottled:" + throttleEvent.isThrottled.toString() + ",expiryTimeStamp:" + throttleEvent.expiryTimeStamp.toString());
+
+// insert throttleevent into the map if it is throttled other wise remove the throttle key it from the throttledata map
+
+public function onReceiveThrottleEvent(ThrottledRequest throttleEvent) {
+    printDebug(KEY_THROTTLE_UTIL, "Event OutputThrottleStream: throttleKey: " + throttleEvent.policyKey +
+    " ,isThrottled:" + throttleEvent.isThrottled.toString() );
     if (throttleEvent.isThrottled) {
-        throttleDataMap[throttleEvent.throttleKey] = throttleEvent;
+        if(throttleEvent.policyKey.length() > 0  ){
+        throttleDataMap[throttleEvent.policyKey] = throttleEvent;
```

Comment:
Fix indentation and formatting issues. You can use 'Ballerina format "file name"' command.
```suggestion
            throttleDataMap[throttleEvent.policyKey] = throttleEvent;
```